src/components/Autosuggest/Autosuggest.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useCallback, useEffect, useRef } from "react";
+import {
+  View,
+  TextInput,
+  FlatList,
+  Text,
+  TouchableNativeFeedback,
+  Keyboard
+} from "react-native";
+import PropTypes from "prop-types";
+
+import styles from "./styles";
+
+const valuePropType = {
+  id: PropTypes.number,
+  label: PropTypes.string,
+  value: PropTypes.string
+};
+
+const Autosuggest = ({ onChangeText, onSelect, items, value, ...props }) => {
+  const [isOpened, setOpen] = useState(false);
+  const [collection, setCollection] = useState(items);
+
+  const inputRef = useRef();
+
+  const hideList = useCallback(() => {
+    setOpen(false);
+    inputRef.current.blur();
+  }, [isOpened]);
+
+  useEffect(() => {
+    Keyboard.addListener("keyboardDidHide", hideList);
+    return () => Keyboard.removeListener("keyboardDidHide", hideList);
+  }, [isOpened]);
+
+  const onSelectValue = useCallback(
+    val => {
+      onSelect(val);
+      setCollection(items);
+      Keyboard.dismiss();
+      setOpen(false);
+    },
+    [isOpened]
+  );
+
+  const onChangeValue = useCallback(
+    text => {
+      onChangeText(text);
+      setCollection(
+        items.filter(
+          item => item.value.toLowerCase().indexOf(text.toLowerCase()) !== -1
+        )
+      );
+    },
+    [collection, onChangeText, value]
+  );
+
+  const handleFocus = useCallback(() => setOpen(true), [isOpened]);
```

Comment:
do we really need to optimize this simple line with `setOpen`? 🙂
also, shouldn't we pass `isOpened` to the `setOpen`?

Reply:
@DimatiBart yes, without `useCallback` these functions will be recreated on every render, on every click 🙂